### DISCUSSION TITLE
[hints.md] Confusing hints by `Callable` and `Generator`

### DISCUSSION THREAD

**Discussion by F-park** [laike9m/Python-Type-Challenges#84] [Q&A] [answered]
现在 `hints.md` 里的 `Callable` 指向的是 `https://docs.python.org/3/library/collections.abc.html#collections.abc.Callable`
![image](https://github.com/laike9m/Python-Type-Challenges/assets/52167622/1a409505-63a7-4605-a17b-97f25be1b3d3)

但 `https://docs.python.org/3/library/typing.html#annotating-callable-objects` 中有详细的例子，指向这个地址是否会更好
![image](https://github.com/laike9m/Python-Type-Challenges/assets/52167622/ea72a911-7709-4b3a-bba9-d91995ea489c)

---

 `Generator` 也有类似的问题，`typing` 的文档里有例子。
`https://docs.python.org/3/library/collections.abc.html#collections.abc.Generator` -> `https://docs.python.org/3/library/typing.html#typing.Generator`

**Comment by laike9m** [comment, marked as answer]
这是有意的，因为
> Deprecated since version 3.9: [collections.abc.Callable](https://docs.python.org/3/library/collections.abc.html#collections.abc.Callable) now supports subscripting ([]). See [PEP 585](https://peps.python.org/pep-0585/) and [Generic Alias Type](https://docs.python.org/3/library/stdtypes.html#types-genericalias).

![image](https://github.com/laike9m/Python-Type-Challenges/assets/2592205/f0bc9556-7ff5-4c0d-830f-594748d14bfe)
